File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math

def solution(progresses, speeds):
    answer = []
    day=[]
        
    for i in range(len(progresses)):
        day.append(math.ceil((100-progresses[i])/speeds[i]))
            
    stk=[]
    answer=[]
    
    for i in day:
        if stk and stk[0] < i:
            answer.append(len(stk))
            stk.clear()
            
        stk.append(i)
            
    if stk:
        answer.append(len(stk))
        
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <string>
#include <vector>
#include <queue>
#include <iostream>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    vector <int> days;
    queue<int> q;
    for(auto i=0; i<progresses.size(); i++){
        days.push_back((100-progresses[i]+speeds[i]-1)/speeds[i]);
    }
    
    for(auto a: days){
        if(q.empty() || q.front()>=a){
            //동일할때
            q.push(a);
        } 
        else{
            int cnt=0;
            while(!q.empty()){
                q.pop();
                cnt++;
            }
            answer.push_back(cnt);
            q.push(a);
        }
    }
    if(!q.empty()){
        answer.push_back(q.size());
    }
    

    return answer;
}